source/application/AMDatamanAppController.cpp

```diff
@@ -242,10 +242,12 @@ bool AMDatamanAppController::startup() {
 			return AMErrorMon::errorAndReturn(this, AMDATAMANAPPCONTROLLER_STARTUP_ERROR_HANDING_NON_FIRST_TIME_USER, "Problem with Acquaman startup: handling non-first-time user.");
 	}
 
-
 	if(!startupRegisterDatabases())
 		return AMErrorMon::errorAndReturn(this, AMDATAMANAPPCONTROLLER_STARTUP_ERROR_REGISTERING_DATABASES, "Problem with Acquaman startup: registering databases.");
 
+	if(!startupDatabaseUpgrades())
```

Comment:
- yes, it covers both first time and every time cases.
- As you pointed below as well, the reason I put this is after `startUpRegisterDatabases()` is that I want to use method `if (!facility->storeToDb(databaseToUpgrade_))`, which can save the thumbnail at the same time.